src/main/java/org/broadinstitute/hellbender/tools/HaplotypeCallerSpark.java

```diff
@@ -133,6 +135,13 @@ public boolean requiresReference(){
 
     @Override
     protected void runTool(final JavaSparkContext ctx) {
+        //TODO remove me when https://github.com/broadinstitute/gatk/issues/4274 is fixed
+        if(hcArgs.emitReferenceConfidence == ReferenceConfidenceMode.GVCF
+                && (AbstractFeatureReader.hasBlockCompressedExtension(output) || output.endsWith(IOUtil.BCF_FILE_EXTENSION))) {
+            throw new UserException.UnimplementedFeature("It is currently not possible to write a compressed g.vcf or bcf.gz from HaplotypeCallerSpark.  " +
+                                            "See https://github.com/broadinstitute/gatk/issues/4274 for more details.");
```

Comment:
Are you repeating this check so that the tool can fail early, or would the check in `VariantSparkSink` not catch this?

Either way, my comment above also applies here: `Doesn't this forbid writing any BCFs when in GVCF mode (not just bcf.gz)? Was this the intent? Both the error message and the ticket #4274 seem incomplete, since they don't mention disallowing g.bcf`

Reply:
Repeating it to catch it early, otherwise it will run the entire haplotype caller and then fail.  Updated here as well.